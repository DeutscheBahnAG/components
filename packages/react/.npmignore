# Documentation
*-doc.js
*-doc.scss
*-doc.css
# Tests
*.snap
*.test.*
#
.release-it.js
tsconfig.build.json
wss_config.yml
