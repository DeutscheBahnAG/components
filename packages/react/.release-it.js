module.exports = {
  git: {
    commitMessage: 'chore: Release @bahn-x/react@${version}',
    requireBranch: 'main',
    tagName: '@bahn-x/react@${version}',
  },
  npm: {
    publishPath: 'dist',
  },
};
