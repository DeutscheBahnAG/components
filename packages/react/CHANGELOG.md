# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

Make sure to format this file using Prettier.

## [Unreleased]

## [1.1.1] - 2022-06-28

### Fixed

- Require latest design fixes of `@db-design/styles`

## [1.1.0] - 2022-06-28

### Added

- Columns support for Container

## [1.0.1] - 2022-05-11

### Fixed

- Missing inline spacer for `<Track>`

## [1.0.0] - 2022-05-10

### Changed

- Renamed `mt`/`mb` to `logoUnitsTop`/`logoUnitsBottom` for `<Logo>` and `<Pulse>`

## [1.0.0-pre.24] - 2022-03-23
