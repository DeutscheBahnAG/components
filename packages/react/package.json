{
  "name": "@db-design/react",
  "version": "1.0.0-pre.19",
  "description": "DB React Components",
  "repository": {
    "directory": "packages/react",
    "url": "git@ssh.git.tech.rz.db.de:bahn-x/webdev/web.git",
    "type": "git"
  },
  "license": "SEE LICENSE IN LICENSE.md",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "dependencies": {
    "@db-design/react-icons": "^1.0.0-pre.1",
    "@db-design/styles": "^1.0.0-pre.17",
    "clsx": "^1.1.1",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-modal": "3.12.1"
  },
  "devDependencies": {
    "@testing-library/react": "^12.1.2",
    "@types/react-modal": "^3.10.6",
    "release-it": "^14.2.2",
    "typescript": "^4.1.3"
  },
  "scripts": {
    "build": "rm -rf dist && tsc --project tsconfig.build.json",
    "postbuild": "./postbuild.sh",
    "release": "release-it --disable-metrics"
  },
  "sideEffects": [
    "./dist/button/*",
    "./dist/modal/*",
    "./dist/pulse/*",
    "./dist/triptime/*",
    "./dist/triptimespan/*"
  ]
}
