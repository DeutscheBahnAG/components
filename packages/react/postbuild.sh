#! /bin/bash

cp -p package.json dist &&
for f in `ls *.md`; do cp -p $f dist; done &&
for f in `ls src/*.d.ts`; do cp -p $f dist; done &&
for f in `ls src/*/README.md`; do cp -p $f ${f/src/dist}; done &&
perl -i -pe 's#dist/index#index#' dist/package.json
perl -i -pe 's#dist/module/index#module/index#' dist/package.json
