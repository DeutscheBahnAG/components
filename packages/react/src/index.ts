export * as TransportchipIcons from './transportchip/icons';

export { default as Anchor } from './anchor';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as Container } from './container';
export { default as Copy } from './copy';
export { default as Layout } from './layout';
export { default as Link } from './link';
export { default as Loadingindicator } from './loadingindicator';
export { default as Logo } from './logo';
export { default as Modal } from './modal';
export { default as Notification } from './notification';
export { default as Passwordfield } from './passwordfield';
export { default as Pulse } from './pulse';
export { default as Select } from './select';
export { default as Status } from './status';
export { default as Textfield } from './textfield';
export { default as Theme } from './theme';
export { default as Time } from './time';
export { default as Title } from './title';
export { default as Track } from './track';
export { default as Transportchip } from './transportchip';
export { default as Triptime } from './triptime';
export { default as Triptimespan } from './triptimespan';
export { default as Radiobutton } from './radiobutton';
export { default as Accordion } from './accordion';
export { default as Accordionitem } from './accordionitem';
export { default as List } from './list';
