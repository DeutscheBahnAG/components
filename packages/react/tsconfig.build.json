{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "esModuleInterop": true,
    "jsx": "react",
    "lib": ["dom", "ES2020"],
    "noImplicitAny": false,
    "outDir": "dist",
    "target": "ES2015",
    "sourceMap": false
  },
  "include": ["src"],
  "exclude": ["**/*.test.*", "**/*-doc.*", "**/*.snap"]
}
