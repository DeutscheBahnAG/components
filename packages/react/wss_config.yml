#DISCLAIMER
#All sensible data is not persisted and deleted shortly after usage. So be assured we cope according to any law with your data!

# NOTE: see here how these configs are used https://git.tech.rz.db.de/ass/dependencychecker/whitesource/whitesource-dockered-scanner/-/blob/master/Docker/base/wrapper/scan

##################
#   WHITESOURCE  #
##################
wss_project_name="db-design-react"
wss_project_version="$CI_COMMIT_BRANCH"
  #wss_product_version="latest" DO NOT SET THIS IF USING wss_project_version as this overrides it

  # The following variables must be set as env variables in the Gitlab bahn-x group, they are valid for all subgroups
  #
  # wss_apiKey  -> your organization key, from Whitesource WebGUI:  Integrate > Organization > API key
  # wss_userKey -> CI tech-user key, as received initially via email, also found in 1pass

  # The following variables must be set as env variables in the Gitlab bahn-x subgroups like TMP, Analytics etc
  #
  # wss_product_name  -> set set this to one of our whitesource "products"  https://whitesource.int-app.comp.db.de/Wss/WSS.html#!allProducts
  # wss_product_token -> the products key, from Whitesource WebGUI:  Integrate > Product Tokens > Token

  # Npm specific config: If you have scoped dependencies, please add them according to the conventions below:
  # npmScopedDeps="@yourScope:registry=yourRegistryLink;@yourScope2:registry=yourRegistryLink2"
npmScopedDeps="@bahn-x:registry=https://npm.pkg.github.com"

  # Whitesource will scan your entire project by deafult.
  # Change this, if you want scan only a specific path.
wss_scan_folder=packages/react/

# You feel brave today?
# Great, use your own wss-unified-agent.config file as of here: https://whitesource.atlassian.net/wiki/spaces/WD/pages/489160834/Unified+Agent+Configuration+File+Parameters
# Tell us, where we will find it in your repo structure, starting from the root directory.
# no leading slash!!
#
# !! Please note that in this case all information (including project / product name and version) must be specified in this team-specific Whitesource config file and will no longer be read from wss_config.yml
# (apart from wss_scan_folder and wss_javaParams which is meta-information (i.e. not Whitesource specific) and therefore still needs to be specified).
#wss_specific_team_config=whitesource/wss-unified-agent.config
