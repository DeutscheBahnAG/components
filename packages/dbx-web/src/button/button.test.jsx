import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from './button';

describe('Button Component', () => {
  it('should render a button', () => {
    const { container } = render(<Button>My button 1</Button>);
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('My button 1');
    expect(container).toMatchSnapshot();
  });

  it('should render a submit button', () => {
    const { container } = render(<Button type={Button.types.SUBMIT}>My button 2</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should render a secondary button', () => {
    const { container } = render(<Button variant={Button.variants.SECONDARY}>My button 3</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should render a full-width button', () => {
    const { container } = render(<Button fullWidth>My button 4</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should render a button with addional attributes', () => {
    const { container } = render(<Button title="Click me">My button 5</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should render a button with addional classes', () => {
    const { container } = render(<Button className="my-class">My button 6</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should render a button when `href` is not set', () => {
    const { container } = render(<Button>My button 7</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should render an anchor when `href` is set', () => {
    const { container } = render(<Button href="#">My button 8</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should render a button with custom `aria-label`', () => {
    const { container } = render(
      <Button aria-label="This is a custom aria-label">My button 9</Button>
    );
    expect(container).toMatchSnapshot();
  });
});
