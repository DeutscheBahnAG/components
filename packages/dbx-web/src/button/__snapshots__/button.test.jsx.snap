// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render a button 1`] = `
<div>
  <button
    class="dbx-button dbx-button--primary dbx-button--default dbx-button--l"
    type="button"
  >
    My button 1
  </button>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Button Component should render a button when \`href\` is not set 1`] = `
<div>
  <button
    class="dbx-button dbx-button--primary dbx-button--default dbx-button--l"
    type="button"
  >
    My button 7
  </button>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Button Component should render a button with addional attributes 1`] = `
<div>
  <button
    class="dbx-button dbx-button--primary dbx-button--default dbx-button--l"
    title="Click me"
    type="button"
  >
    My button 5
  </button>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Button Component should render a button with addional classes 1`] = `
<div>
  <button
    class="dbx-button dbx-button--primary dbx-button--default dbx-button--l my-class"
    type="button"
  >
    My button 6
  </button>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Button Component should render a button with custom \`aria-label\` 1`] = `
<div>
  <button
    aria-label="This is a custom aria-label"
    class="dbx-button dbx-button--primary dbx-button--default dbx-button--l"
    type="button"
  >
    My button 9
  </button>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Button Component should render a full-width button 1`] = `
<div>
  <button
    class="dbx-button dbx-button--primary dbx-button--default dbx-button--block dbx-button--l"
    type="button"
  >
    My button 4
  </button>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Button Component should render a secondary button 1`] = `
<div>
  <button
    class="dbx-button dbx-button--secondary dbx-button--default dbx-button--l"
    type="button"
  >
    My button 3
  </button>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Button Component should render a submit button 1`] = `
<div>
  <button
    class="dbx-button dbx-button--primary dbx-button--default dbx-button--l"
    type="submit"
  >
    My button 2
  </button>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Button Component should render an anchor when \`href\` is set 1`] = `
<div>
  <a
    class="dbx-button dbx-button--primary dbx-button--default dbx-button--l"
    href="#"
  >
    My button 8
  </a>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;
